src/server/worker/pipeline/service/service.go

```diff
@@ -74,60 +70,30 @@ func Run(driver driver.Driver, logger logs.TaggedLogger) error {
 // The given context will be canceled if a newer commit is ready, then this will
 // wait for the previous callback to return before calling the callback again
 // with the latest commit.
-func forEachCommit(pachClient *client.APIClient, pipelineInfo *pps.PipelineInfo, logger logs.TaggedLogger, cb func(context.Context, *pfs.CommitInfo) error) error {
+func forEachPipelineJob(pachClient *client.APIClient, pipelineInfo *pps.PipelineInfo, logger logs.TaggedLogger, cb func(context.Context, *pps.PipelineJobInfo) error) error {
 	// These are used to cancel the existing service and wait for it to finish
 	var cancel func()
 	var eg *errgroup.Group
-	return pachClient.SubscribeCommit(
-		client.NewRepo(pipelineInfo.Pipeline.Name),
-		"",
-		"",
-		pfs.CommitState_READY,
-		func(ci *pfs.CommitInfo) error {
-			if cancel != nil {
-				logger.Logf("canceling previous service, new commit ready")
-				cancel()
-				if err := eg.Wait(); err != nil && !errors.Is(err, context.Canceled) {
-					return err
-				}
+	return pachClient.SubscribePipelineJob(pipelineInfo.Pipeline.Name, true, func(pji *pps.PipelineJobInfo) error {
+		if cancel != nil {
+			logger.Logf("canceling previous service, new commit ready")
+			cancel()
+			if err := eg.Wait(); err != nil && !errors.Is(err, context.Canceled) {
+				return err
 			}
-			logger.Logf("starting new service, commit: %s", ci.Commit.ID)
-			var ctx context.Context
-			ctx, cancel = context.WithCancel(pachClient.Ctx())
-			eg, ctx = errgroup.WithContext(ctx)
-			eg.Go(func() error { return cb(ctx, ci) })
-			return nil
-		},
-	)
-}
-
-func ensurePipelineJob(pachClient *client.APIClient, pipeline string, commit *pfs.Commit, logger logs.TaggedLogger) (*pps.PipelineJobInfo, error) {
-	// Check if a job was previously created for this commit. If not, make one
-	pipelineJobInfos, err := pachClient.ListPipelineJob("", nil, commit, -1, true)
-	if err != nil {
-		return nil, err
-	}
-	if len(pipelineJobInfos) > 1 {
-		return nil, errors.Errorf("multiple pipeline jobs found for commit: %s@%s", commit.Branch.Repo.Name, commit.ID)
-	} else if len(pipelineJobInfos) < 1 {
-		pipelineJob, err := pachClient.CreatePipelineJob(pipeline, commit, nil)
-		if err != nil {
-			return nil, err
 		}
-		logger.Logf("created new pipeline job %q for output commit %q", pipelineJob.ID, commit.ID)
-		// get PipelineJobInfo to look up spec commit, pipeline version, etc (if this
-		// worker is stale and about to be killed, the new job may have a newer
-		// pipeline version than the master. Or if the commit is stale, it may
-		// have an older pipeline version than the master)
-		return pachClient.InspectPipelineJob(pipelineJob.ID, false)
-	}
-	// Get latest pipeline job state.
-	logger.Logf("found existing pipeline job %q for output commit %q", pipelineJobInfos[0].PipelineJob.ID, commit.ID)
-	return pachClient.InspectPipelineJob(pipelineJobInfos[0].PipelineJob.ID, false)
+		logger.Logf("starting new service, commit: %s", pji.OutputCommit.ID)
```

Comment:
not exactly sure who would be looking for this log line - would including the job ID be useful?

Reply:
I mean, the job ID is the same as the output commit ID, but mentioning that it's the job ID is probably more relevant to this code so I'll update the message.